cmd/ddev/cmd/composer-create.go

```diff
@@ -43,9 +44,22 @@ ddev composer create --preserve-flags --no-interaction psr/log
 		osargs := []string{}
 		if len(os.Args) > 3 {
 			osargs = os.Args[3:]
-			osargs = nodeps.RemoveItemFromSlice(osargs, "--yes")
-			osargs = nodeps.RemoveItemFromSlice(osargs, "-y")
+			// I don't know why, but cmd.Flags().GetBool("preserve-flags") is always false, so we can check it here
+			preserveFlags = slices.Contains(osargs, "--preserve-flags")
```

Comment:
This baffles me, I don't understand why Cobra reads `--preserve-flags` as `false` no matter how I try to pass this flag.

Anyway, at least `--preserve-flags` now works as described (it didn't before).